### ISSUE TITLE
login – incompatible with loopback-pinentry

### ISSUE THREAD

**Issue by grawity** [keybase/client#1303]
I use `pinentry-mode loopback` in my gnupg.conf, so that I'd get the simple "Enter passphrase:" prompt instead of a popup window, and this seems to be incompatible with keybase:

```
Choose a key: 1
(a few seconds pass)
▶ ERROR GPG error: exit status 2
```

Maybe `--pinentry-mode default` should be added, or the loopback mode documented as not supported?

Arch Linux aur/keybase-release 4428.3e4cfb2-1

**Comment by oconnor663** [comment]
I'm trying to repro, but in the meantime what happens if you run keybase in standalone mode? You'll need to do a `keybase ctl stop` to stop the daemon, and then `keybase --standalone login.

**Comment by grawity** [comment]
In standalone mode, I see the prompt in the same terminal (as expected with loopback mode):

```
10:23:21.077237 ▶ [DEBU keybase user.go:426] 130 - VerifySelfSig via SigChain
10:23:21.077289 ▶ [DEBU keybase gpg_cli.go:262] 131 | running Gpg: /usr/bin/gpg2 [--armor --export-secret-key 2357e10cef4f7ed27e233ad5d24f6cb2c1b52632]
Enter passphrase: 
```

I forgot to attach debug output in the regular daemon mode:

```
10:24:02.145673 ▶ [DEBU keybase log.go:30] 139 | running Gpg: /usr/bin/gpg2 [--armor --export-secret-key 2357e10cef4f7ed27e233ad5d24f6cb2c1b52632]
10:24:02.146119 ▶ [DEBU keybase log.go:30] 13a - VerifySelfSig via SigChain
10:24:02.158761 ▶ [INFO keybase log.go:32] 13b gpg: cannot open '/dev/tty': No such device or address
10:24:02.159476 ▶ [DEBU keybase log.go:30] 13c an error occurred during provisioning, logging out
```

**Comment by maxtaco** [comment]
To see why we nee the secret key, checkout `keybase help gpg`.  We need the secret key to sign statements, and we're hoping going forward to not shell though GPG as it's error-prone (c.f., this issue). 

Are you on GPG v2.1?

**Comment by maxtaco** [comment]
As for the secret key import issue, see #1308.  It's a feature we want to roll out, but we'll poll users for how needed it is, to see how we can prioritize it.

**Comment by grawity** [comment]
Yeah, it's 2.1.9. I don't really have large problems regarding `gpg --export-key`, it still seems more reliable than the alternativel of hand-parsing pubring.kbx ...

Anyway I can switch back to a regular pinentry for now.

**Comment by oconnor663** [comment]
Unfortunately it looks like `--pinentry-mode` is a gpg2.1-only feature. We'd have to check the version and set that flag only for 2.1.

@maxtaco maybe the most general fix for this would be to have the client invoke gpg to export the key, as an additional RPC on SecretUI or something like that? Not sure if it's worth the complexity.
